In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

# Save locally
df.to_csv("titanic.csv", index=False)

print("Dataset downloaded successfully!")

Dataset downloaded successfully!


In [3]:
import pandas as pd

def analyze_dataset():
    # Load dataset
    df = pd.read_csv("titanic.csv")

    # Display top rows
    print("\n--- Top 5 Rows ---")
    print(df.head())

    # Find column with highest value
    numeric_cols = df.select_dtypes(include=['number'])
    max_values = numeric_cols.max()
    highest_column = max_values.idxmax()

    print("\n--- Column with Highest Value ---")
    print(f"{highest_column} (Max Value: {max_values[highest_column]})")

    # Count missing values
    print("\n--- Missing Values Count ---")
    print(df.isnull().sum())

    # Insights
    print("\n--- Insights ---")

    # 1. Dataset shape
    print(f"1. Dataset has {df.shape[0]} rows and {df.shape[1]} columns.")

    # 2. Survival rate
    survival_rate = df["Survived"].mean() * 100
    print(f"2. Survival Rate: {survival_rate:.2f}%")

    # 3. Average age
    avg_age = df["Age"].mean()
    print(f"3. Average Age: {avg_age:.2f}")

    # 4. Gender survival comparison
    gender_survival = df.groupby("Sex")["Survived"].mean()
    print("4. Survival rate by gender:")
    print(gender_survival)

    # 5. Class survival comparison
    class_survival = df.groupby("Pclass")["Survived"].mean()
    print("5. Survival rate by passenger class:")
    print(class_survival)


analyze_dataset()


--- Top 5 Rows ---
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500

**Insights:**

The dataset contains information about passengers such as age, gender, class, and survival status.

The overall survival rate is around 38%, indicating most passengers did not survive.

Females had a significantly higher survival rate compared to males.

Passengers in 1st class had a higher survival rate than those in lower classes.

The dataset contains missing values, especially in the Age and Cabin columns, which may require data cleaning.